src/front/glsl/functions.rs

```diff
@@ -281,4 +281,29 @@ impl Program {
         f.fill_global_use(self.module.global_variables.len(), &self.module.functions);
         f
     }
+
+    pub fn declare_function(&mut self, f: Function) -> Result<(), ErrorKind> {
+        let name = f
+            .name
+            .clone()
+            .ok_or_else(|| ErrorKind::SemanticError("Unnamed function".into()))?;
+        let stage = self
+            .entry_points
+            .iter()
+            .find_map(|(n, s)| if *name == *n { Some(*s) } else { None });
```

Comment:
no need to `find`, since it's a hashmap now. You can just do `self.entry_points.get(&name)`

Reply:
damn, you spot everything 😄